lib/src/main/java/com/quiltdata/quiltcore/Entry.java

```diff
@@ -0,0 +1,89 @@
+package com.quiltdata.quiltcore;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.quiltdata.quiltcore.key.PhysicalKey;
+
+import software.amazon.awssdk.utils.BinaryUtils;
+
+
+public class Entry {
+    public static enum HashType {
+        SHA256,
+    }
+
+    public static class Hash {
+        public final HashType type;
+        public final String value;
+
+        public Hash(HashType type, String value) {
```

Comment:
What's the recommended way to check for null values? E.g., is a single condition ok:

```
if (type == null || value == null) {
  throw new NullPointerException();
}
```

Or should it be per-variable, with a detail message?

Or should it be `IllegalArgumentException` rather than `NullPointerException`?

I'm also curious about `@NonNull` annotations - but from what I understand, they're just hints for the IDE, and have no effect otherwise?